graphql_compiler/schema/__init__.py

```diff
@@ -301,26 +301,33 @@ def _parse_date_value(value):
     return arrow.get(value, "YYYY-MM-DD").date()
 
 
-def _serialize_datetime(value):
+def _serialize_datetime(value: Any) -> str:
```

Comment:
Do we also want to enforce no tzinfo in the `validate_argument_type` function in `query_formatting/common.py`? I know deserialization can't produce a datetime with tzinfo, but not all arguments come from deserialization. They could be hard-coded or generated in python code, or come from a custom deserializer.

Reply:
I think that it makes sense that tzinfo is None for the following reasons:
1. It is easier to stop enforcing it in the future than to start enforcing it.
2. Without enforcing this, it would not be safe to start paginating previously un-paginated queries since pagination currently removes timezone info. 
3.  There would be no way to fulfill the docstring of `validate_argument_type` that the validation done in that function is stricter than the validation done in another other backend. 

Overall the way I look at it is that this gives us some nice guarantees that make the code easier to maintain, so I would prefer to keep enforce this until we have concrete reasons not to.

Reply:
I think you misunderstood what I'm saying: Let's add extra validation in `validate_argument_type`, checking that `tzinfo is None`.

Reply:
Oh my bad. We already have this validation though:
```
    elif is_same_type(GraphQLDateTime, stripped_type):
        if not isinstance(value, (datetime.date, arrow.Arrow)):
            _raise_invalid_type_error(name, (datetime.date, arrow.Arrow), value)
        try:
            GraphQLDateTime.serialize(value)
        except ValueError as e:
            raise GraphQLInvalidArgumentError(e)
```
```
def _serialize_datetime(value: Any) -> str:
    """Serialize a DateTime object to its proper ISO-8601 representation."""
    # Python datetime.datetime is a subclass of datetime.date, but in this case, the two are not
    # interchangeable. Rather than using isinstance, we will therefore check for exact type
    # equality.
    #
    # We don't allow Arrow objects as input since it seems that Arrow objects are always tz aware.
    # This is supported by the fact that the `.naive` Arrow method returns a datetime object instead
    # of an Arrow object.
    if type(value) == datetime and value.tzinfo is None:
        return value.isoformat()
    else:
        raise ValueError(
            f"Expected a timezone naive datetime object. Got {value} of type {type(value)} instead."
        )
```